src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/FhirRequestContextRouteDataPopulatingFilterAttribute.cs

```diff
@@ -67,16 +79,17 @@ public override void OnActionExecuting(ActionExecutingContext context)
                 }
             }
 
-            // Set the resource type based on the route data
-            RouteData routeData = context.RouteData;
+            base.OnActionExecuting(context);
+        }
 
-            if (routeData?.Values != null)
-            {
-                if (routeData.Values.TryGetValue(KnownActionParameterNames.ResourceType, out object resourceType))
-                {
-                    fhirRequestContext.ResourceType = resourceType?.ToString();
-                }
-            }
+        public override void OnActionExecuted(ActionExecutedContext context)
+        {
+            // The result can either be a FhirResult or an OperationOutcomeResult which both extend BaseActionResult.
+            var result = context.Result as IResourceActionResult;
+
+            _fhirRequestContextAccessor.FhirRequestContext.ResourceType = result?.GetResultTypeName();
+
```

Comment:
It seems like we have two different methods being used in this filter for populating the resource type, is it possible for them to be different within the same service call?

Reply:
It is possible. For example, create/read/update, they operate on a single resource type and is identical for both request and response. However, in the case of search, the incoming request could indicate that it's a search for Patient, but the response is Bundle.